atlite/wind.py

```diff
@@ -90,3 +102,158 @@ def extrapolate_wind_speed(ds, to_height, from_height=None):
                           "units" : "m s**-1"})
 
     return wnd_spd.rename(to_name)
+
+
+# Cache
+_oedb_turbines = None
+
+def download_turbineconf(turbine, store_locally=True):
+    """Download a windturbine configuration from the OEDB database.
+    
+    Download the configuration of a windturbine model from the OEDB database
+    into the local 'windturbine_dir'.
+    The OEDB database can be viewed here:
+    https://openenergy-platform.org/dataedit/view/supply/turbine_library
+    (2019-07-22)
+    Only one turbine configuration is downloaded at a time, if the 
+    search parameters yield an ambigious result, no data is downloaded.
+    
+    Parameters
+    ----------
+    turbine : dict
+        Search parameters, either provide the turbine model id (takes priority)
+        or a manufacturer and turbine name, e.g. the following are identical:
+        {'id':10}, {'id':10, name:'E-53/800', 'manufacturer':'Unknown'},
+        {name:'E-53/800', 'manufacturer':'Enercon'}
+    store_locally : bool
+        (Default: True) Whether the downloaded config should be stored locally
+        in config.windturbine_dir.
+        
+    Returns
+    -------
+    turbineconf : dict
+        The turbine configuration downloaded and stored is also returned as a dict.
+        Has the same format as returned by 'atlite.ressource.get_turbineconf(name)'.
+    """
+
+    ## Parse information of different allowed 'turbine' values
+    parsed = False
+    # Assume id
+    if isinstance(turbine, int):
+        turbine = {'id': turbine}
+        parsed = True
+    elif isinstance(turbine,str):
+        s = turbine
+        turbine = {}
+        # Clean the string
+        s = s.strip()
+        s = s.replace('oedb:','')
+        
+        # 'turbine' is just a str(-inged) id
+        m = re.match("(^\d+$)", s)
+        if m:
+            turbine['id'] = turbine.get('id', int(m[1]))
+            s = "" # Consume the string after success
+            parsed = True
+
+        # 'turbine' is a name or combi of manufacturer + name
+        # Matches e.g. "TurbineName", "Manu1/Manu2[_ ]Turbine/with/number"
+        m = re.match("(?P<manufacturer>[-a-zA-Z0-9\/]*?)?(?:[\s_]*)?(?P<name>[-a-zA-Z0-9\/]+)$", s)
+        if m:
+            turbine['name'] = turbine.get('name', m.group('name'))
+            turbine['manufacturer'] = turbine.get('manufacturer', m.group('manufacturer'))
```

Comment:
netither `"name"` nor `"manufacturer"` can be in turbine at this point.

Reply:
As a side-note, instead of a pattern like
```
turbine['name'] = turbine.get('name', m.group('name'))
```
better to use the dedicated function
```
turbine.setdefault('name', m.group('name'))
```

Reply:
I guess I prefer a more gentle way of setting k:v paris in `dict` ;-)

Side note: Right, forgot about `setdefault(...)` .